src/main/java/com/mraof/minestuck/entity/ai/attack/GroundSlamGoal.java

```diff
@@ -0,0 +1,104 @@
+package com.mraof.minestuck.entity.ai.attack;
+
+import com.mraof.minestuck.entity.AttackingAnimatedEntity;
+import com.mraof.minestuck.entity.animation.PhasedMobAnimation;
+import com.mraof.minestuck.util.MSSoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.PathfinderMob;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.AABB;
+
+import java.util.List;
+
+/**
+ * Alternative attack for when the attack target is out of standard range, results in the target being flung into the air if they were on the ground and not crouching.
+ */
+public class GroundSlamGoal<T extends AttackingAnimatedEntity> extends AnimatedAttackWhenInRangeGoal<T>
+{
+	public static final int GROUP_SLAM_COOLDOWN = 20; //how long it should take before another nearby mob can ground slam
+	
+	public GroundSlamGoal(T entity, PhasedMobAnimation animation, float minRange, float maxRange, int actionCooldown)
+	{
+		super(entity, animation, minRange, maxRange, actionCooldown);
+	}
+	
+	@Override
+	public boolean canUse()
+	{
+		return super.canUse() && entity.hasFinishedCooldown();
+	}
+	
+	@Override
+	public void start()
+	{
+		super.start();
+		
+		Level level = this.entity.getLevel();
+		AABB aabb = new AABB(this.entity.blockPosition()).inflate(4);
+		List<AttackingAnimatedEntity> entityList = level.getEntitiesOfClass(AttackingAnimatedEntity.class, aabb);
+		if(!entityList.isEmpty())
+		{
+			for(AttackingAnimatedEntity iteratedEntity : entityList)
+			{
+				//TODO Figure out if it can be made so that any entities with the GroundSlamGoal recieve the cooldown, not just entities of the same type
+				if(iteratedEntity != this.entity && iteratedEntity.getType() == this.entity.getType() && !iteratedEntity.existingCooldownIsLonger(GROUP_SLAM_COOLDOWN))
+				{
+					iteratedEntity.setCooldown(GROUP_SLAM_COOLDOWN + phasedAnimation.getTotalAnimationLength()); //plays at beginning to prevent overlapping
+				}
+			}
+		}
+	}
+	
+	@Override
+	public void attemptToLandAttack(PathfinderMob attacker, LivingEntity target)
+	{
+		Level level = entity.getLevel();
+		level.playSound(null, this.entity.blockPosition(), MSSoundEvents.ENTITY_SLAM.get(), SoundSource.HOSTILE, 1, 1);
+		
+		//flinging any nearby smaller mobs on the ground indiscriminately
+		AABB aabb = new AABB(attacker.blockPosition()).inflate(4);
+		List<LivingEntity> entityList = level.getEntitiesOfClass(LivingEntity.class, aabb);
+		if(!entityList.isEmpty())
+		{
+			for(LivingEntity iteratedEntity : entityList)
+			{
+				boolean smallerThanAttacker = iteratedEntity.getBoundingBox().getSize() < attacker.getBoundingBox().getSize();
+				
+				if(iteratedEntity != attacker && iteratedEntity != target && iteratedEntity.isOnGround() && !iteratedEntity.noPhysics && smallerThanAttacker)
```

Comment:
Should the distance also be checked here, perhaps?

Reply:
Im not sure I am seeing why, since the AABB already defines a similar area of distance. Or am I misunderstanding?

Reply:
Well, it's not entirely consistent with how distances work with the target. The AABB is also currently using some arbitrary and possibly different size.

Reply:
That is true, do you feel like their arbitrary relationship is too much?

Reply:
It's a problem that I would want fixed. But I think it would be fine with a todo for now since it's not worth the time before 4/13.

Reply:
It hath been done